### ISSUE TITLE
Restrict modification of specifications already in use

### ISSUE THREAD

**Issue by santiagobernalg** [boa-project/boa-admin#12]
If there are records associated with the specification in edition, special care must be taken; For this they have the following rules:
Fields can be deleted from it.
Optional fields may be added.
It can add mandatory fields as long as you assign default values for the DCOs that use the specification.
It can change the visibility of the fields.
The specification can be deleted as long as there are no objects in the system that make use of it.

**Comment by davidherney** [comment]
The specifications can be evolve. In order to change in the time, the specifications require a version and the specification version require be saved in the objects metadata. If a specification change, an object metadata is updated to the new specification version only when the object metadata is edited.
@jesusoterove Agree?

**Comment by jesusoterove** [comment]
Absolutely, we cannot prevent specs from evolve, in fact, when there are specs updates (new versions) and an Object using old specs is opened for metadata edition, we should notify the users, that a new version of the spec is available, so they should decide whether they want to upgrade the Object to the new spec or not.
